traptor/traptor.py

```diff
@@ -657,7 +657,9 @@ def _find_rule_matches(self, tweet_dict):
             # User mentions
             if 'user_mentions' in tweet_dict['entities']:
                 for tag in tweet_dict['entities']['user_mentions']:
-                    query = query + " " + tag['id_str'].encode("utf-8")
+                    id_str = tag.get('id_str')
+                    if id_str:
+                        query = query + " " + id_str.encode("utf-8")
```

Comment:
Any chance for an encoding exception? Perhaps, after looking at the rest of this monster method, the caller of `_find_rule_matches()` needs to wrap it's call in a try/except and gracefully move along if a single tweet fails in here. That's my recommendation.